chanzuckerberg/single-cell-data-portal
# feat(cellguide): added marker gene mode in the ontology view

## Reason for Change

- #5524 

## Changes

- added new output to the computational marker gene pipeline which contains the reformatted computational marker genes (`marker_gene_presence.json.gz`). This JSON has structure `gene --> organism --> tissue --> (cell types + stats)[]`
- added new marker gene mode to the ontology view. clicking on any 'tree' icon in the computational marker gene table will scroll to the ontology view and display all cell types that share that marker gene, colored in black. the tooltip for black nodes now display marker gene stats. a tag appears on the top-right, which lets users deactivate marker gene mode.
- reimplemented the legend in css and updated the legend for marker gene mode. removed the no-longer applicable hidden terms square and added a legend item for the green node (which indicates the current cell type). moved the full-screen button to the top right.
- reimplemented nodes to be div elements wrapped in `foreignObject` tags. replaced StyledCircle/StyledRect with StyledPie. Circles are now StyledPie's with degree set to 360.
- changed "no descendants" visual

## Screenshots + Demo
The tree icon appears on hover.
<img width="158" alt="image" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/16548075/316333d0-f94f-42d2-afe2-7ccde0b2755e">

Marker gene mode:
<img width="1042" alt="image" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/16548075/33e35704-2992-4ef7-8a46-7737c9e5c9b0">

New no descendants visual:
<img width="201" alt="image" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/16548075/679f8d53-fa22-40a0-8d43-6fe3477d433c">

New legend format:
<img width="410" alt="image" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/16548075/f530bac1-aa65-436b-bf1e-200ec3f5d16f">


## Testing steps

- [RDEV](https://cmg-mode-frontend.rdev.single-cell.czi.technology/cellguide)
- Updated unit test
- Added e2e test for the new marker gene mode in the ontology view
- Ran a test run of the computational marker gene pipeline and yielded expected results
